DEVELOPER_GUIDE.md

```diff
@@ -303,6 +303,9 @@ The artifact will include extension settings for the sample Hello World extensio
   opensearchPort: 9200
 ```
 
+You can optionally add `routeNamePrefix:` as a value to the yml. This setting allows you to prefix all your registered NamedRoute names.
```

Comment:
@DarshitChanpura Is this going back to the notion of an abbreviation for an extension? i.e. Anomaly Detection's abbreviation is `ad`? If that's the case, then can we call this `extensionAbbreviation` or similar? 

What does it mean for this to be optional? For security, if an admin wants to grant full permissions to a user to use ad extensions would that be granting them `ad:*` cluster_permissions or something different?

Could the prefix be the abbreviation if it is set and otherwise take the name of the extension? Maybe support for both `ad:*` and `anomaly-detection:*`? FYI I think `:`, `/` and `*` should be forbidden in the abbreviation. These are used as delimiters in (transport) action names

Reply:
They can use this as a prefix or define their own when registering routes. I'll change the permission to be only alphanumeric and can include '-' and '_'. I prefer to call it a routeNamePrefix because that is what it does. It won't always be an abbreviation. Maybe in future iterations we enforce wrapping all routes with this prefix, but as of now it is left upto the developers whether they would like to use it or not.

Reply:
>  I'll change the permission to be only alphanumeric and can include '-' and '_'. 

The `-` will run afoul of the NamedRoute check.